bundles/org.eclipse.equinox.cm.test/src/org/eclipse/equinox/cm/test/Activator.java

```diff
@@ -13,52 +13,31 @@
  *******************************************************************************/
 package org.eclipse.equinox.cm.test;
 
+import java.util.Map;
+import org.eclipse.osgi.container.ModuleContainer;
 import org.osgi.framework.*;
-import org.osgi.service.packageadmin.PackageAdmin;
+import org.osgi.framework.namespace.IdentityNamespace;
+import org.osgi.framework.wiring.FrameworkWiring;
+import org.osgi.resource.Namespace;
+import org.osgi.resource.Requirement;
 
-public class Activator implements BundleActivator {
+public class Activator {
 
-	private static PackageAdmin packageAdmin;
-	private static BundleContext bundleContext;
-	private ServiceReference<PackageAdmin> packageAdminRef;
-
-	public void start(BundleContext context) throws Exception {
-		packageAdminRef = context.getServiceReference(PackageAdmin.class);
-		setPackageAdmin(context.getService(packageAdminRef));
-		setBundleContext(context);
-	}
-
-	public void stop(BundleContext context) throws Exception {
-		setBundleContext(null);
-		setPackageAdmin(null);
-		context.ungetService(packageAdminRef);
-	}
-
-	private static synchronized void setBundleContext(BundleContext context) {
-		bundleContext = context;
-	}
-
-	private static synchronized void setPackageAdmin(PackageAdmin service) {
-		packageAdmin = service;
+	static BundleContext getBundleContext() {
+		return FrameworkUtil.getBundle(Activator.class).getBundleContext();
 	}
 
-	static synchronized BundleContext getBundleContext() {
-		return bundleContext;
-	}
+	public static Bundle getBundle(String symbolicName) {
+		FrameworkWiring fkw = getBundleContext().getBundle(Constants.SYSTEM_BUNDLE_LOCATION)
```

Comment:
This is basically what `Platform.getBundle(String)` is doing. Do you think it would be ok for this test to require o.e.core.runtime?